### PR TITLE
[RFC] vim-patch:8.1.0053 first argument of 'completefunc' has inconsistent type

### PR DESCRIPTION
**Pull request by cangscop**
Problem:    The first argument given to 'completefunc' can be Number or
            String, depending on the value.
Solution:   Avoid guessing the type of an argument, use typval_T in the
            callers of call_vim_function(). (Ozaki Kiichi)
vim/vim#2993

### CODE REVIEW

**Comment by justinmk** [comment]
shoudl be `vim-patch:8.1.0053` (which is a valid Vim git tag) not `vim-patch:8.1.53` (which is not a valid Vim git tag). `scripts/vim-patch.sh` would do this for you next time :)

File: src/nvim/eval.c [hunk 1]
```diff
@@ -1275,116 +1275,80 @@ int get_spellword(list_T *const list, const char **ret_word)
...
 // Return OK or FAIL.
 int call_vim_function(
     const char_u *func,
     int argc,
-    const char_u *const *const argv,
-    bool safe,                       // use the sandbox
-    int str_arg_only,               // all arguments are strings
+    typval_T *argv,
+    int safe,                       // use the sandbox
     typval_T *rettv
```
Comment (justinmk) [inline src/nvim/eval.c]: in the vim patch, `safe` is the last arg. what's the reason for changing the order? 

Also `safe` could stay as `bool`.

File: src/nvim/eval.c [hunk 2]
```diff
@@ -1275,116 +1275,80 @@ int get_spellword(list_T *const list, const char **ret_word)
 
 
 // Call some vim script function and return the result in "*rettv".
-// Uses argv[argc] for the function arguments.  Only Number and String
-// arguments are currently supported.
+// Uses argv[argc-1] for the function arguments. argv[argc]
+// should have type VAR_UNKNOWN.
```
Comment (justinmk) [inline src/nvim/eval.c]: the comment is different in the vim patch. why change it?

**Comment by janlazo** [comment]
vim-patch:8.1.0055 changes were included in this PR so the patch is N/A now.

**Pull request merged**